curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md

```diff
@@ -16,7 +16,7 @@ for (var i = 0; i < arr.length; i++) {
 }
 ```
 
-Remember that Arrays have zero-based numbering, which means the last index of the array is length - 1. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops when <code>i</code> is at length - 1.
+Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops when <code>i</code> is at <code>length - 1</code>.
```

Comment:
```suggestion
Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops the loop when <code>i</code> is equal to <code>length</code>.  In this case the last iteration has <code>i === 4</code> and outputs <code>6</code> to the console.
```
How about this way of splitting up @thecodingaviator's version?  I thought mentioning what the loop does in the final iteration might help clear up any lingering confusion about indices.
@organicdude Arrays are introduced as [being zero-based](https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/basic-javascript/access-array-data-with-indexes/) so I think it's fine to repeat that here.

Reply:
Building on your suggestion, I came up with this:
```suggestion
Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops the loop when <code>i</code> is equal to <code>length</code>.  In this case the last iteration is <code>i === 4</code> i.e. when <code>i</code> becomes equal to <code>arr.length</code> and outputs <code>6</code> to the console.
```